File: C Sharp/Database/CatalogSubreport.cs
using System;

namespace Aspose.Cells.Demos
{
    /// <summary>
    /// Summary description for CatalogSubreport.
    /// </summary>
    public class CatalogSubreport : DbBase
    {
        public CatalogSubreport(string path)
            : base(path)
        {

        }

        public Workbook CreateCatalogSubreport()
        {
            try
            {
                DBInit();

                //Open the connection object
                this.oleDbConnection1.Open();
                //Specify an SQL as command text
                this.oleDbSelectCommand1.CommandText = "SELECT ProductName, ProductID, QuantityPerUnit, UnitPrice FROM Products ORDER BY " +
                    "ProductName";
                //Fill a datatable
                this.oleDbDataAdapter1.Fill(this.dataTable1);
            }
            catch
            {
            }
            finally
            {
                if (this.oleDbDataAdapter1 != null)
                    this.oleDbDataAdapter1.Dispose();
                if (this.oleDbConnection1 != null)
                    this.oleDbConnection1.Close();
            }

            //Open a template file
	    string designerFile = MapPath("~/Designer/Northwind.xls");
        Workbook workbook = new Workbook(designerFile);
  
            //Get the sheet
            Worksheet sheet = workbook.Worksheets["Sheet3"];
            //Name the sheet
            sheet.Name = "Catalog Subreport";
            //Get the cells in the sheet
            Cells cells = sheet.Cells;
            //Import the datatable to the sheet
            cells.ImportDataTable(this.dataTable1, false, 0, 1);

            //Remove the unnecessary worksheets in the workbook
            for (int i = 0; i < workbook.Worksheets.Count; i++)
            {
                sheet = workbook.Worksheets[i];
                if (sheet.Name != "Catalog Subreport")
                {
                    workbook.Worksheets.RemoveAt(i);
                    i--;
                }

            }
            //Retrun the generated workbook
            return workbook;
        }
    }
}

File: Visual Basic/Database/CatalogSubreport.vb
Imports Microsoft.VisualBasic
Imports System

Namespace Aspose.Cells.Demos
	''' <summary>
	''' Summary description for CatalogSubreport.
	''' </summary>
	Public Class CatalogSubreport
		Inherits DbBase
		Public Sub New(ByVal path As String)
			MyBase.New(path)

		End Sub

		Public Function CreateCatalogSubreport() As Workbook
			Try
				DBInit()

				'Open the connection object
				Me.oleDbConnection1.Open()
				'Specify an SQL as command text
				Me.oleDbSelectCommand1.CommandText = "SELECT ProductName, ProductID, QuantityPerUnit, UnitPrice FROM Products ORDER BY " & "ProductName"
				'Fill a datatable
				Me.oleDbDataAdapter1.Fill(Me.dataTable1)
			Catch
			Finally
				If Me.oleDbDataAdapter1 IsNot Nothing Then
					Me.oleDbDataAdapter1.Dispose()
				End If
				If Me.oleDbConnection1 IsNot Nothing Then
					Me.oleDbConnection1.Close()
				End If
			End Try

			'Open a template file
		Dim designerFile As String = MapPath("~/Designer/Northwind.xls")
		Dim workbook As New Workbook(designerFile)

			'Get the sheet
			Dim sheet As Worksheet = workbook.Worksheets("Sheet3")
			'Name the sheet
			sheet.Name = "Catalog Subreport"
			'Get the cells in the sheet
			Dim cells As Cells = sheet.Cells
			'Import the datatable to the sheet
			cells.ImportDataTable(Me.dataTable1, False, 0, 1)

			'Remove the unnecessary worksheets in the workbook
			Dim i As Integer = 0
			Do While i < workbook.Worksheets.Count
				sheet = workbook.Worksheets(i)
				If sheet.Name <> "Catalog Subreport" Then
					workbook.Worksheets.RemoveAt(i)
					i -= 1
				End If

				i += 1
			Loop
			'Retrun the generated workbook
			Return workbook
		End Function
	End Class
End Namespace


